### PR TITLE
Uses Litani build system in CBMC proofs

### PR DESCRIPTION
**Pull request by feliperodri**
*Issue #, if available:*

N/A.

*Description of changes:*

This PR makes CBMC proofs get built with the Litani build tool (hosted [here](https://github.com/awslabs/aws-build-accumulator)).

The only really interesting file in this PR is the `run-cbmc-proofs.py` script at the bottom. Apart from that, this PR bumps the template submodule pointer, introduces the litani submodule, sets a few variables for litani, and makes a small change to each proof Makefile.

This PR supersedes PR #677.   

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: verification/cbmc/proofs/run-cbmc-proofs.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+This tool automates the process of running `make report` in each of
+the CBMC proof directories. The tool calculates the dependency graph
+of all tasks needed to build, run, and report on all the proofs, and
+executes these tasks in parallel.
+
+The tool is roughly equivalent to doing this:
+
+        litani init --project "my-cool-project";
+
+        find . -name cbmc-proof.txt | while read -r proof; do
```
Comment (karkhaz) [inline verification/cbmc/proofs/run-cbmc-proofs.py:31]: change `cbmc-proof.txt` to `cbmc-batch.yaml` please

**Review by karkhaz — CHANGES_REQUESTED** [review]

**Comment by karkhaz** [comment]
Thank you! Just one small comment.

File: verification/cbmc/proofs/run-cbmc-proofs.py (hunk 1 above)
Comment (feliperodri) [reply verification/cbmc/proofs/run-cbmc-proofs.py:31]: Implemented as suggested.

File: verification/cbmc/proofs/run-cbmc-proofs.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env python3
```
Comment (nrdg42) [inline verification/cbmc/proofs/run-cbmc-proofs.py:1]: This file needs different permissions. Please run `chmod u+x ./run-cbmc-proofs.py` and update your PR

Comment (feliperodri) [reply verification/cbmc/proofs/run-cbmc-proofs.py:1]: implemented as suggested.

**Review by karkhaz — APPROVED** [review]

**Pull request merged**